packages/core/server/admin-api/services/bulk-generate.ts

```diff
@@ -108,65 +105,68 @@ const generateUrlAliases = async (parms: GenerateParams) => {
+    // Get all relations for the type
     await Promise.all(languages.map(async (lang) => {
-      const urlPattern = await getPluginService('urlPatternService').findByUid(type, lang);
-      const languageRelations = getPluginService('urlPatternService').getRelationsFromPattern(urlPattern);
+      const urlPatterns = await getPluginService('urlPatternService').findByUid(type, lang);
+      const languageRelations = getPluginService('urlPatternService').getRelationsFromPattern(urlPatterns);
       relations = [...relations, ...languageRelations];
     }));
 
-    // Query all the entities of the type that do not have a corresponding URL alias.
+    // Query all the entities of the type that do not have a corresponding URL alias
     const entities = await strapi.entityService.findMany(type, {
-      filters: {
-        url_alias: null,
-      },
+      filters: { url_alias: null },
       locale: 'all',
-      // @ts-ignore
-      populate: {
-        ...relations.reduce((obj, key) => ({ ...obj, [key]: {} }), {}),
-      },
+      populate: { ...relations.reduce((obj, key) => ({ ...obj, [key]: {} }), {}) },
     });
 
-    /**
-     * @todo
-     * We should do a Promise.all(entities.map()) here to speed up the process.
-     * Using that method we can create all the URL aliases in parallel.
-     * Currently this is not possible due to the duplicateCheck function.
-     * Race conditions can occur when two entities have the same URL path.
-     */
-    // For all those entities we will create a URL alias and connect it to the entity.
-    // eslint-disable-next-line no-restricted-syntax
-    for (const entity of entities) {
-      // @ts-ignore
-      // eslint-disable-next-line no-await-in-loop, @typescript-eslint/no-unsafe-argument
-      const urlPattern = await getPluginService('urlPatternService').findByUid(type, entity.locale);
-      const generatedPath = getPluginService('urlPatternService').resolvePattern(type, entity, urlPattern);
-
-      // eslint-disable-next-line no-await-in-loop
-      const newUrlAlias = await getPluginService('urlAliasService').create({
-        url_path: generatedPath,
-        generated: true,
-        contenttype: type,
-        // @ts-ignore
-        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-        locale: entity.locale,
-      });
-
-      // eslint-disable-next-line no-await-in-loop
-      await strapi.entityService.update(type, entity.id, {
-        data: {
-          // @ts-ignore
-          url_alias: newUrlAlias.id,
-        },
-      });
-
-      generatedCount += 1;
-    }
+    // Get URL patterns for the type
+    const urlPatterns = await getPluginService('urlPatternService').findByUid(type);
+
+    // Loop through each entity and generate paths
+    await Promise.all(entities.map(async (entity) => {
+      const resolvedPathsArray = await Promise.all(
+        // eslint-disable-next-line @typescript-eslint/require-await
+        urlPatterns.map(async (urlPattern) => {
```

Comment:
This doesn't have to be an async function as there is no await in it.
By removing the `async` part you can remove the `eslint-disable` comment as well as remove the `await Promise.all(` surrounding this `.map` as there are no promises to await.